warehouse/query-core/src/main/java/datawave/query/iterator/logic/AndIterator.java

```diff
@@ -256,17 +258,30 @@ public void seek(Range range, Collection<ByteSequence> columnFamilies, boolean i
             try {
                 for (NestedIterator<T> itr : child.leaves()) {
                     if (itr instanceof SeekableIterator) {
-                        ((SeekableIterator) itr).seek(range, columnFamilies, inclusive);
+                        try {
+                            ((SeekableIterator) itr).seek(range, columnFamilies, inclusive);
+                        } catch (Exception e2) {
+                            if (itr.isIndexOnly()) {
```

Comment:
Should this be isNonEventField() instead?  non-event fields (which include index-only) require index lookups as well .

Reply:
NonEventField is much better than IndexOnly, thanks for pointing that out